client/src/main/scala/org/http4s/netty/client/Http4sHandler.scala

```diff
@@ -57,6 +61,12 @@ private[netty] class Http4sHandler[F[_]](cb: Http4sHandler.CB[F], dispatcher: Di
         super.channelRead(ctx, msg)
     }
 
+  override def channelInactive(ctx: ChannelHandlerContext): Unit = {
+    cb(Right(Resource.pure(Response(Status.RequestTimeout))))
```

Comment:
not sure if we should do this, raise an exception or cancel the Resource.

Reply:
Do we need to guard against double invocation of `cb`? Chances are good it protects itself but it would be good to verify.
Wrt the failure mode, we didn't necessarily get a response from the server so I don't think we should return a `Response` which would signify that the server did send us _something_. Maybe we should return an `Exception`, something like a `ClosedException`? It's also not clear that it's a timeout exception: we don't know why the channel was closed.

Reply:
agreed. I have an alternative impl which will fix a few other issues. Will close this and reimplement the fix there.